src/content/app/species-selector/SpeciesSelector.tsx

```diff
@@ -30,13 +31,14 @@ import SpeciesSelectorGeneSearchView from './views/species-selector-gene-search-
 import styles from './SpeciesSelector.module.css';
 
 const SpeciesSelector = () => {
+  const { pathname } = useLocation();
+  const [searchParams] = useSearchParams();
   const modalView = useAppSelector(getSpeciesSelectorModalView);
   const dispatch = useAppDispatch();
 
-  const shouldShowResultsView = [
-    'species-search',
-    'popular-species-genomes'
-  ].includes(modalView || '');
+  const shouldShowResultsView =
+    pathname.includes('/search') &&
+    (searchParams.has('query') || searchParams.has('species_taxonomy_id'));
```

Comment:
This should go into the component that matches the `/search` route.

Reply:
✅